src/main.rs

```diff
@@ -63,12 +63,12 @@ fn work_valid(root: [u8; 32], work: [u8; 8], difficulty: u64) -> (bool, u64) {
     (result_difficulty >= difficulty, result_difficulty)
 }
 
-fn to_multiplier(difficulty: u64) -> f64 {
-    ((-Wrapping(LIVE_DIFFICULTY)).0 as f64) / ((-Wrapping(difficulty)).0 as f64)
+fn to_multiplier(difficulty: u64, base_difficulty: u64) -> f64 {
+    ((-Wrapping(base_difficulty)).0 as f64) / ((-Wrapping(difficulty)).0 as f64)
 }
 
-fn from_multiplier(multiplier: f64) -> u64 {
-    -Wrapping(((-Wrapping(LIVE_DIFFICULTY)).0 as f64) / (multiplier)).0 as u64
+fn from_multiplier(multiplier: f64, base_difficulty: u64) -> u64 {
+    -Wrapping(((-Wrapping(base_difficulty)).0 as f64) / (multiplier)).0 as u64
```

Comment:
These are probably better written as (this example for from_multipler):
```rs
((base_difficulty.wrapping_neg() as f64) / multiplier) as u64).wrapping_neg()
```
You can explicitly specify wrapping for individual operations instead of as a type, though they'll compile to the same thing.